ddtrace/contrib/openai/patch.py

```diff
@@ -244,6 +247,72 @@ def record_usage(self, span, usage):
             span.set_metric("openai.response.usage.%s_tokens" % token_type, num_tokens)
             self._statsd.distribution("tokens.%s" % token_type, num_tokens, tags=tags)
 
+    def generate_completion_llm_records(self, resp, span, args, kwargs):
+        # type: (Any, Span, List[Any], Dict[str, Any]) -> None
+        """Generate payloads for the LLM Obs API from a completion."""
+        if not self._config.llmobs_enabled:
+            return
+        choices = resp.choices
+        n = kwargs.get("n", 1)
+        prompt = kwargs.get("prompt", "")
+        if isinstance(prompt, str):
+            prompt = [prompt]
+        # Note: LLMObs ingest endpoint only accepts a 1:1 prompt-response mapping per record,
+        #  so we need to deduplicate and send unique prompt-response records if n > 1.
+        for i in range(n):
+            unique_choices = choices[i::n]
+            attrs_dict = {
+                "type": "completion",
+                "id": resp.id,
+                "timestamp": resp.created * 1000,
```

Comment:
Can this value be None?

Reply:
Nope, these are always guaranteed to be not-None values by the openai API.